src/libraries/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -69,7 +69,7 @@ internal static partial class PathInternal
         /// </summary>
         internal static bool IsValidDriveChar(char value)
         {
-            return (value >= 'A' && value <= 'Z') || (value >= 'a' && value <= 'z');
+            return (uint)((value | 0x20) - 'a') <= (uint)('z' - 'a');
```

Comment:
Can this use the new API, or is it too low level to be dependent on it?

Reply:
Unfortunately this file is compiled into a library that has a pre-.NET 7 target.  It wasn't worth ifdef'ing it to use the new API in some builds and the open-coded version in others.